middle_end/flambda2/simplify/apply_cont_rewrite.ml

```diff
@@ -264,19 +268,27 @@ let rewrite_exn_continuation rewrite id exn_cont =
   in
   let _, extra_args1 =
     let extra_args_list =
-      List.map2
-        (fun (arg : EA.t) extra_param ->
-          match arg with
-          | Already_in_scope simple -> simple, Bound_parameter.kind extra_param
-          | New_let_binding _ | New_let_binding_with_named_args _ ->
-            (* Note: this is unsupported for now. If we choose to support it in
-               the future, we must take care of not introducing a wrapper
-               continuation, which would come with its own pushtrap/poptrap. *)
-            Misc.fatal_error
-              "[New_let_binding] are currently forbidden for exn continuation \
-               rewrites")
-        (extra_args_list rewrite id)
-        (Bound_parameters.to_list rewrite.extra_params)
+      match extra_args_list rewrite id with
+      | Invalid ->
+        (* CR gbury: This is not supported for now, but adding support for it
+           should be relatively easy and straight-forward *)
+        Misc.fatal_error "[Invalid]"
```

Comment:
Maybe we should add a better error message

Reply:
done